src/ios/SentryCordova.m

```diff
@@ -8,121 +8,245 @@
+      [scope clearBreadcrumbs];
     }];
+  }];
 }
 
+- (void)setUser:(CDVInvokedUrlCommand *)command {
+  [self.commandDelegate runInBackground:^{
+    NSDictionary *user = [command.arguments objectAtIndex:0];
+    NSDictionary *otherUserKeys = [command.arguments objectAtIndex:1];
 
-- (void)setUserContext:(CDVInvokedUrlCommand *)command {
-    [self.commandDelegate runInBackground:^{
-        id arg = [command.arguments objectAtIndex:0];
-        if ([arg isKindOfClass:NSDictionary.class]) {
-            SentryClient.sharedClient.user = [SentryJavaScriptBridgeHelper createSentryUserFromJavaScriptUser:arg];
+    [SentrySDK configureScope:^(SentryScope *_Nonnull scope) {
+      if (nil == user && nil == otherUserKeys) {
+        [scope setUser:nil];
+      } else {
+        SentryUser *userInstance = [[SentryUser alloc] init];
+
+        if (nil != user) {
+          [userInstance setUserId:user[@"id"]];
+          [userInstance setEmail:user[@"email"]];
+          [userInstance setUsername:user[@"username"]];
+        }
+
+        if (nil != otherUserKeys) {
+          [userInstance setData:otherUserKeys];
         }
+
+        [scope setUser:userInstance];
+      }
     }];
+  }];
 }
 
-- (void)setTagsContext:(CDVInvokedUrlCommand *)command {
-    [self.commandDelegate runInBackground:^{
-        if (SentryClient.sharedClient && SentryClient.sharedClient.tags) {
-            id dict = [command.arguments objectAtIndex:0];
-            if ([dict isKindOfClass:NSDictionary.class]) {
-                NSMutableDictionary *newDict = [NSMutableDictionary new];
-                [newDict addEntriesFromDictionary:SentryClient.sharedClient.tags];
-                [newDict addEntriesFromDictionary:dict];
-                SentryClient.sharedClient.tags = newDict;
-            }
-        }
+- (void)setExtra:(CDVInvokedUrlCommand *)command {
+  [self.commandDelegate runInBackground:^{
+    NSString *key = [command.arguments objectAtIndex:0];
+    NSString *extra = [command.arguments objectAtIndex:1];
+
+    [SentrySDK configureScope:^(SentryScope *_Nonnull scope) {
+      [scope setExtraValue:extra forKey:key];
     }];
+  }];
 }
 
-- (void)setExtraContext:(CDVInvokedUrlCommand *)command {
-    [self.commandDelegate runInBackground:^{
-        if (SentryClient.sharedClient && SentryClient.sharedClient.extra) {
-            id dict = [command.arguments objectAtIndex:0];
-            if ([dict isKindOfClass:NSDictionary.class]) {
-                NSMutableDictionary *newDict = [NSMutableDictionary new];
-                [newDict addEntriesFromDictionary:SentryClient.sharedClient.extra];
-                [newDict addEntriesFromDictionary:dict];
-                SentryClient.sharedClient.extra = newDict;
-            }
-        }
+- (void)setContext:(CDVInvokedUrlCommand *)command {
+  [self.commandDelegate runInBackground:^{
+    NSString *key = [command.arguments objectAtIndex:0];
+    NSDictionary *context = [command.arguments objectAtIndex:1];
```

Comment:
Could be a follow up PR after releasing a preview but shouldn't we use some defensive programming here and check in fact we have 2 items to read from `arguments` and that they are indeed of the type we assume they are?